From 6d8e8e3ab4f6c14fb65ee6b1b4f0f4f9a8f3e9f7 Mon Sep 17 00:00:00 2001
From: suckbluefrog <suckbluefrog@proton.me>
Date: Fri, 10 Jul 2026 00:00:00 +000
Subject: [PATCH] libsanitizer: Remove scc from libsanitizer

Some modern kernel header sets no longer provide linux/scc.h. GCC's
libsanitizer still includes it only to expose sizes for the obsolete SCC
types, while the ioctl interceptor table does not reference SCC ioctls.

Drop the stale include and the related size metadata so libsanitizer can
build against those headers.

Upstream: Pending
Signed-off-by: suckbluefrog <suckbluefrog@proton.me>
---
 .../sanitizer_platform_limits_posix.cpp       | 3 ---
 .../sanitizer_platform_limits_posix.h         | 2 --
 2 files changed, 5 deletions(-)

diff --git a/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cpp b/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cpp
index 7bf1fc0f5d2..6cbb0d84666 100644
--- a/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cpp
+++ b/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cpp
@@ -153,7 +153,6 @@
 #include <linux/lp.h>
 #include <linux/mroute.h>
 #include <linux/mroute6.h>
-#include <linux/scc.h>
 #include <linux/serial.h>
 #include <sys/msg.h>
 #include <sys/ipc.h>
@@ -523,8 +522,6 @@ namespace __sanitizer {
   unsigned struct_kbsentry_sz = sizeof(struct kbsentry);
   unsigned struct_mtconfiginfo_sz = sizeof(struct mtconfiginfo);
   unsigned struct_nr_parms_struct_sz = sizeof(struct nr_parms_struct);
-  unsigned struct_scc_modem_sz = sizeof(struct scc_modem);
-  unsigned struct_scc_stat_sz = sizeof(struct scc_stat);
   unsigned struct_serial_multiport_struct_sz
       = sizeof(struct serial_multiport_struct);
   unsigned struct_serial_struct_sz = sizeof(struct serial_struct);
diff --git a/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h b/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h
index 009b2cce6c0..2e7d7e73361 100644
--- a/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h
+++ b/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h
@@ -1058,8 +1058,6 @@
 extern unsigned struct_kbsentry_sz;
 extern unsigned struct_mtconfiginfo_sz;
 extern unsigned struct_nr_parms_struct_sz;
-extern unsigned struct_scc_modem_sz;
-extern unsigned struct_scc_stat_sz;
 extern unsigned struct_serial_multiport_struct_sz;
 extern unsigned struct_serial_struct_sz;
 extern unsigned struct_sockaddr_ax25_sz;
-- 
2.50.1
